### PR TITLE
Fix #32

### LINKED ISSUE #32: QGIS crash when closing the panel after layer deletion
ProfileTool 4.1.7
QGIS dev and QGIS 3.8

After plotting a dem, if the dem is deleted from QGIS and the ProfileTool panel is closed without removing the dem plotted from it, QGIS crash. Tested on several rasters with the same effect.

### PR DESCRIPTION
**Pull request by nicogodet**
QGIS crash when closing the panel after layer deletion

### CODE REVIEW

**Comment by nicogodet** [comment]
#45

**Comment by JaviBecerra** [comment]
Thanks for the pull request @nicogodet. Roberto will merge and publish a new version next week,
FYI, we looked onto this issue a couple times last year, your fix works but to handle layer deletion properly takes a bit more of work that we could not fully accomplish any of both times; looking back I see that at that moment we should have gone with your fast fix and call it a day until we had the occasion to properly refactor that part of the plugin, sorry about that.
Regards,
Javier

**Comment by nicogodet** [comment]
Thanks @jeremyk6 for the quick fix proposed in #32 

It may not be the best way to fix this issue but it avoids qgis to crash. 
I use this fix since I create this pull request without any problem.

Thank @JaviBecerra for your answer.
Regards,

**Comment by nicogodet** [comment]
@JaviBecerra I think this last commit is a good improvement but still perfectible.

File: tools/tableviewtool.py [hunk 1]
```diff
@@ -106,7 +106,7 @@ def addLayer(self , iface, mdl, layer1 = None):
             if ok :
                 choosenBand = int(testqt) - self.bandoffset
             else:
-                return 2
+                choosenBand = 2
```
Comment (nicogodet) [inline tools/tableviewtool.py:109]: Maybe I miss something..

**Pull request closed without merging**